### PR TITLE
Update submodules to the latest d2 only versions

### PR DESCRIPTION
**Pull request by matthias-wende-sociomantic**
Since the D2 migration was completed this upgrades the dhtnode to use the latest major D2 only versions for all libraries.

### CODE REVIEW

**Comment by matthias-wende-sociomantic** [comment]
Most of the work is done.

The missing bits are:
* the integration tests have to be updated
* unfortunately I have found bugs in swarm and dhtproto that have to go into a patch release before this PR can be merged
* I'm going to split the submodule update commit and squash the new commits with the related fixes.

**Comment by matthias-wende-sociomantic** [comment]
This is ready for review now!

**Comment by matthias-wende-sociomantic** [comment]
According to travis I was too fast :(. Integration test(s) are still failing.

**Comment by joseph-wakeling-frequenz** [comment]
This LGTM -- I've confirmed that it builds & passes all integration & unittests (for both devel and production builds) for every patch.

**Comment by don-clugston-sociomantic** [comment]
This is hard to review, because it's doing so much.
Could the earlier commits (those which compile without the actual submodule update commits) be split into a separate PR?

**Comment by joseph-wakeling-frequenz** [comment]
I reckon most of them can be split into separate standalone patch PRs.

**Comment by joseph-wakeling-frequenz** [comment]
BTW, the `Remove no_delay` patch should probably be retitled, it doesn't remove the option from the app, just stops it being used (albeit it likely was a no-op anyway).

**Comment by don-clugston-sociomantic** [comment]
I don't think it's necessary to split each patch into a separate PR, I just think the "stop using deprecated names" commits should be split from the "update submodules" commits.

Additionally, my initial reaction is that the big "update submodules" commit seems really big. If that's inevitable, and it can't be broken up, then it definitely should be in its own PR.

**Comment by joseph-wakeling-frequenz** [comment]
> I just think the "stop using deprecated names" commits should be split from the "update submodules" commits.

Which change exactly are you referring to here?  Unfortunately some of the transitions (e.g. from `add` to `addHandler`) are not so easy to do as standalone changes (the 14.x protocol versions don't really provide adequate support for them, or at least that was the case for d{mq,ls}proto).

**Comment by joseph-wakeling-frequenz** [comment]
All of the new major releases (ocean, swarm, and dhtproto) come with some hard breaking changes, so it's not possible to do a clean decoupling from the submodule updates -- AFAICS @matthias-wende-sociomantic has done a pretty good job of separating out what _can_ be separated.

**Comment by matthias-wende-sociomantic** [comment]
> BTW, the Remove no_delay patch should probably be retitled, it doesn't remove the option from the app, just stops it being used (albeit it likely was a no-op anyway).

Good catch! It should be removed completely.

**Comment by matthias-wende-sociomantic** [comment]
I've split all possible commits in several smaller PRs which all merged by now. I hope this makes this one easier to review. Unfortunately it's not possible to split the outstanding commit.

File: src/dhtnode/node/RequestHandlers.d [hunk 1]
```diff
@@ -39,24 +38,14 @@ public ConnectionHandler.RequestMap requests;
...
+    requests.addHandler!(GetHashRangeImpl_v0)();
+    requests.addHandler!(PutImpl_v0)();
+    requests.addHandler!(ExistsImpl_v0)();
+    requests.addHandler!(GetImpl_v0)();
+    requests.addHandler!(UpdateImpl_v0)();
+    requests.addHandler!(MirrorImpl_v0)();
+    requests.addHandler!(GetAllImpl_v0)();
+    requests.addHandler!(GetChannelsImpl_v0)();
+    requests.addHandler!(RemoveImpl_v0)();
+    requests.addHandler!(RemoveChannelImpl_v0)();
```
Comment (joseph-wakeling-frequenz) [inline src/dhtnode/node/RequestHandlers.d]: You don't actually need the parentheses around the template parameter here (or for the runtime parameters, for that matter), but this is just FYI, no need to change anything.

**Pull request merged**